File: swift_reference/Views/Onboarding/DedicationCeremonyScreen.swift
import SwiftUI

struct DedicationCeremonyScreen: View {
    let gratitudeNote: String?
    let habitName: String
    let habitCategory: HabitCategory
    let onComplete: () -> Void

    @State private var showVerse: Bool = false
    @State private var showGratitudeTile: Bool = false
    @State private var showHabitTile: Bool = false
    @State private var showButton: Bool = false
    @State private var isDedicated: Bool = false
    @State private var showPulse: Bool = false
    @State private var showFinalMessage: Bool = false
    @State private var glowRadius: CGFloat = 100
    @State private var glowOpacity: Double = 0.08
    @State private var tilesGlow: Bool = false
    @State private var breathe: Bool = false

    var body: some View {
        ZStack {
            RadialGradient(
                colors: [
                    TributeColor.golden.opacity(glowOpacity),
                    TributeColor.softGold.opacity(glowOpacity * 0.4),
                    Color.clear
                ],
                center: .center,
                startRadius: 0,
                endRadius: glowRadius
            )
            .ignoresSafeArea()
            .animation(.easeInOut(duration: 2.5), value: glowRadius)
            .animation(.easeInOut(duration: 2.5), value: glowOpacity)

            VStack(spacing: 0) {
                ScrollView {
                    VStack(spacing: 32) {
                        if !isDedicated {
                            preOfferingContent
                        } else {
                            postOfferingContent
                        }
                    }
                    .padding(.horizontal, 24)
                    .padding(.top, 24)
                    .padding(.bottom, 32)
                }

                if !isDedicated && showButton {
                    Button {
                        performDedication()
                    } label: {
                        HStack(spacing: 10) {
                            Image(systemName: "hands.sparkles.fill")
                                .font(.subheadline)
                            Text("Offer My Tribute")
                        }
                        .tributeButton()
                    }
                    .padding(.horizontal, 24)
                    .padding(.bottom, 32)
                    .transition(.opacity.combined(with: .move(edge: .bottom)))
                }

                if isDedicated && showFinalMessage {
                    Button {
                        onComplete()
                    } label: {
                        HStack(spacing: 8) {
                            Text("Enter Tribute")
                            Image(systemName: "arrow.right")
                                .font(.subheadline)
                        }
                        .tributeButton()
                    }
                    .padding(.horizontal, 24)
                    .padding(.bottom, 32)
                    .transition(.opacity.combined(with: .move(edge: .bottom)))
                }
            }
        }
        .sensoryFeedback(.success, trigger: isDedicated)
        .onAppear {
            startEntryAnimations()
        }
    }

    @ViewBuilder
    private var preOfferingContent: some View {
        VStack(spacing: 8) {
            Text("Your Tribute")
                .font(.system(.title2, design: .serif, weight: .bold))
                .foregroundStyle(.primary)

            Text("Everything you\u{2019}ve set \u{2014} offered to God.")
                .font(.subheadline)
                .foregroundStyle(.secondary)
                .multilineTextAlignment(.center)
        }
        .opacity(showVerse ? 1 : 0)

        VStack(spacing: 14) {
            habitTile(
                icon: "hands.sparkles.fill",
                name: "Daily Gratitude",
                detail: "Check-in",
                accent: TributeColor.golden,
                isShowing: showGratitudeTile,
                isGlowing: tilesGlow
            )

            habitTile(
                icon: habitCategory.iconName,
                name: habitName,
                detail: habitCategory.rawValue,
                accent: habitCategory == .abstain ? TributeColor.warmCoral : TributeColor.golden,
                isShowing: showHabitTile,
                isGlowing: tilesGlow
            )
        }

        VStack(spacing: 6) {
            Text("\u{201C}Therefore, I urge you, brothers and sisters, in view of God\u{2019}s mercy, to offer your bodies as a living sacrifice, holy and pleasing to God \u{2014} this is your true and proper worship.\u{201D}")
                .font(.system(.subheadline, design: .serif))
                .italic()
                .foregroundStyle(TributeColor.softGold.opacity(0.6))
                .multilineTextAlignment(.center)
            Text("Romans 12:1")
                .font(.caption)
                .foregroundStyle(TributeColor.golden.opacity(0.5))
        }
        .padding(.top, 4)
        .opacity(showVerse ? 1 : 0)
    }

    @ViewBuilder
    private var postOfferingContent: some View {
        Spacer().frame(height: 40)

        ZStack {
            if showPulse {
                DedicationPulseView()
                    .allowsHitTesting(false)
            }

            VStack(spacing: 20) {
                ZStack {
                    Circle()
                        .fill(
                            RadialGradient(
                                colors: [
                                    TributeColor.golden.opacity(0.4),
                                    TributeColor.golden.opacity(0.1)
                                ],
                                center: .center,
                                startRadius: 0,
                                endRadius: 50
                            )
                        )
                        .frame(width: 90, height: 90)
                        .scaleEffect(breathe ? 1.08 : 1.0)
                        .animation(.easeInOut(duration: 2).repeatForever(autoreverses: true), value: breathe)

                    Image(systemName: "hands.sparkles.fill")
                        .font(.system(size: 38))
                        .foregroundStyle(TributeColor.golden)
                }

                Text("Your tribute is set.")
                    .font(.system(.title3, design: .serif, weight: .bold))
                    .foregroundStyle(.primary)

                Text("Go in grace.")
                    .font(.system(.headline, design: .serif))
                    .foregroundStyle(TributeColor.softGold)
            }
            .opacity(showFinalMessage ? 1 : 0)
            .scaleEffect(showFinalMessage ? 1 : 0.92)
        }
        .frame(maxWidth: .infinity)

        VStack(spacing: 6) {
            Text("\u{201C}The steadfast love of the Lord never ceases; his mercies never come to an end; they are new every morning.\u{201D}")
                .font(.system(.subheadline, design: .serif))
                .italic()
                .foregroundStyle(TributeColor.softGold.opacity(0.5))
                .multilineTextAlignment(.center)
            Text("Lamentations 3:22\u{2013}23")
                .font(.caption)
                .foregroundStyle(TributeColor.golden.opacity(0.4))
        }
        .opacity(showFinalMessage ? 1 : 0)
    }

    private func habitTile(icon: String, name: String, detail: String, accent: Color, isShowing: Bool, isGlowing: Bool) -> some View {
        HStack(spacing: 14) {
            ZStack {
                Circle()
                    .fill(
                        RadialGradient(
                            colors: [
                                accent.opacity(isGlowing ? 0.5 : 0.2),
                                accent.opacity(isGlowing ? 0.2 : 0.06)
                            ],
                            center: .center,
                            startRadius: 0,
                            endRadius: 24
                        )
                    )
                    .frame(width: 48, height: 48)

                Image(systemName: icon)
                    .font(.system(size: 22))
                    .foregroundStyle(accent)
            }

            VStack(alignment: .leading, spacing: 4) {
                Text(name)
                    .font(.system(.headline, design: .serif))

                Text(detail)
                    .font(.caption)
                    .foregroundStyle(.secondary)
            }

            Spacer()

            Image(systemName: isGlowing ? "checkmark.circle.fill" : "circle")
                .font(.title3)
                .foregroundStyle(isGlowing ? TributeColor.golden : Color.white.opacity(0.15))
        }
        .padding(16)
        .background(isGlowing ? accent.opacity(0.08) : TributeColor.cardBackground)
        .clipShape(.rect(cornerRadius: 16))
        .overlay(
            RoundedRectangle(cornerRadius: 16)
                .strokeBorder(isGlowing ? accent.opacity(0.3) : TributeColor.cardBorder, lineWidth: 0.5)
        )
        .opacity(isShowing ? 1 : 0)
        .offset(y: isShowing ? 0 : 12)
        .animation(.easeOut(duration: 0.4), value: isGlowing)
    }

    private func startEntryAnimations() {
        withAnimation(.easeOut(duration: 0.6).delay(0.3)) {
            showVerse = true
        }
        withAnimation(.easeOut(duration: 0.5).delay(0.7)) {
            showGratitudeTile = true
        }
        withAnimation(.easeOut(duration: 0.5).delay(1.0)) {
            showHabitTile = true
        }
        withAnimation(.easeOut(duration: 0.4).delay(1.5)) {
            showButton = true
        }
    }

    private func performDedication() {
        withAnimation(.easeInOut(duration: 0.4)) {
            tilesGlow = true
        }

        withAnimation(.easeInOut(duration: 1.5)) {
            glowRadius = 350
            glowOpacity = 0.25
        }

        showPulse = true

        Task {
            try? await Task.sleep(for: .seconds(1.2))

            withAnimation(.easeInOut(duration: 0.6)) {
                isDedicated = true
                showButton = false
            }

            withAnimation(.easeInOut(duration: 1.0)) {
                glowRadius = 200
                glowOpacity = 0.12
            }

            try? await Task.sleep(for: .seconds(0.4))

            withAnimation(.spring(response: 0.6, dampingFraction: 0.8)) {
                showFinalMessage = true
                breathe = true
            }

            try? await Task.sleep(for: .seconds(1.5))
            showPulse = false
        }
    }
}

struct DedicationPulseView: View {
    @State private var animate: Bool = false

    var body: some View {
        ZStack {
            ForEach(0..<4, id: \.self) { index in
                Circle()
                    .fill(
                        RadialGradient(
                            colors: [
                                TributeColor.golden.opacity(0.4),
                                TributeColor.softGold.opacity(0.15),
                                TributeColor.softGold.opacity(0)
                            ],
                            center: .center,
                            startRadius: 0,
                            endRadius: 120
                        )
                    )
                    .scaleEffect(animate ? 3 : 0.2)
                    .opacity(animate ? 0 : 0.6)
                    .animation(
                        .easeOut(duration: 1.5)
                            .delay(Double(index) * 0.2),
                        value: animate
                    )
            }
        }
        .frame(width: 200, height: 200)
        .onAppear {
            animate = true
        }
    }
}

File: lib/presentation/views/onboarding/dedication_ceremony_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import '../../../domain/entities/habit.dart';
import '../../theme/app_theme.dart';

class DedicationCeremonyScreen extends StatefulWidget {
  final String? gratitudeNote;
  final String habitName;
  final HabitCategory habitCategory;
  final String purposeStatement;
  final HabitTrackingType trackingType;
  final double dailyTarget;
  final String targetUnit;
  final VoidCallback onComplete;
  final String? givenName;

  const DedicationCeremonyScreen({
    super.key,
    required this.gratitudeNote,
    required this.habitName,
    required this.habitCategory,
    this.purposeStatement = '',
    this.trackingType = HabitTrackingType.checkIn,
    this.dailyTarget = 1,
    this.targetUnit = '',
    required this.onComplete,
    this.givenName,
  });

  @override
  State<DedicationCeremonyScreen> createState() => _DedicationCeremonyScreenState();
}

class _DedicationCeremonyScreenState extends State<DedicationCeremonyScreen>
    with SingleTickerProviderStateMixin {
  bool _showVerse = false;
  bool _showGratitudeTile = false;
  bool _showHabitTile = false;
  bool _showButton = false;
  bool _isDedicated = false;
  bool _showPulse = false;
  bool _showFinalMessage = false;
  bool _tilesGlow = false;
  double _glowOpacity = 0.08;
  late final AnimationController _breatheController;

  @override
  void initState() {
    super.initState();
    _breatheController = AnimationController(
      vsync: this,
      duration: const Duration(milliseconds: 2000),
    )..repeat(reverse: true);

    Future.delayed(const Duration(milliseconds: 300), () {
      if (mounted) setState(() => _showVerse = true);
    });
    Future.delayed(const Duration(milliseconds: 700), () {
      if (mounted) setState(() => _showGratitudeTile = true);
    });
    Future.delayed(const Duration(milliseconds: 900), () {
      if (mounted) setState(() => _showHabitTile = true);
    });
    Future.delayed(const Duration(milliseconds: 1500), () {
      if (mounted) setState(() => _showButton = true);
    });
  }

  @override
  void dispose() {
    _breatheController.dispose();
    super.dispose();
  }

  void _performDedication() {
    HapticFeedback.mediumImpact();
    setState(() {
      _tilesGlow = true;
      _glowOpacity = 0.25;
      _showPulse = true;
    });

    Future.delayed(const Duration(milliseconds: 1200), () {
      if (!mounted) return;
      setState(() {
        _isDedicated = true;
        _showButton = false;
        _glowOpacity = 0.12;
      });
    });

    Future.delayed(const Duration(milliseconds: 1600), () {
      if (!mounted) return;
      setState(() => _showFinalMessage = true);
    });

    Future.delayed(const Duration(milliseconds: 3100), () {
      if (!mounted) return;
      setState(() => _showPulse = false);
    });
  }

  Color get _accentColor =>
      widget.habitCategory == HabitCategory.abstain ? TributeColor.warmCoral : TributeColor.golden;

  IconData _categoryIcon() {
    switch (widget.habitCategory) {
      case HabitCategory.exercise: return Icons.fitness_center;
      case HabitCategory.scripture: return Icons.menu_book;
      case HabitCategory.rest: return Icons.bedtime;
      case HabitCategory.fasting: return Icons.no_food;
      case HabitCategory.study: return Icons.school;
      case HabitCategory.service: return Icons.volunteer_activism;
      case HabitCategory.connection: return Icons.people;
      case HabitCategory.health: return Icons.favorite;
      case HabitCategory.abstain: return Icons.shield_rounded;
      default: return Icons.auto_awesome;
    }
  }

  @override
  Widget build(BuildContext context) {
    return AnimatedBuilder(
      animation: _breatheController,
      builder: (context, _) {
        return Stack(
          fit: StackFit.expand,
          children: [
            AnimatedContainer(
              duration: const Duration(milliseconds: 1500),
              decoration: BoxDecoration(
                gradient: RadialGradient(
                  colors: [
                    TributeColor.golden.withValues(alpha: _glowOpacity),
                    TributeColor.softGold.withValues(alpha: _glowOpacity * 0.4),
                    Colors.transparent,
                  ],
                  radius: 1.5,
                ),
              ),
            ),
            Column(children: [
              Expanded(
                child: SingleChildScrollView(
                  padding: const EdgeInsets.fromLTRB(24, 24, 24, 32),
                  child: Column(children: [
                    if (!_isDedicated) _preOfferingContent() else _postOfferingContent(),
                  ]),
                ),
              ),
              if (!_isDedicated && _showButton)
                Padding(
                  padding: const EdgeInsets.fromLTRB(24, 0, 24, 32),
                  child: SizedBox(
                    width: double.infinity,
                    child: ElevatedButton.icon(
                      onPressed: _performDedication,
                      icon: const Icon(Icons.volunteer_activism, size: 16),
                      label: const Text('Offer My Tribute',
                          style: TextStyle(fontWeight: FontWeight.w600, fontSize: 16)),
                      style: ElevatedButton.styleFrom(
                        backgroundColor: TributeColor.golden,
                        foregroundColor: TributeColor.charcoal,
                        padding: const EdgeInsets.symmetric(vertical: 16),
                        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(14)),
                      ),
                    ),
                  ),
                ),
              if (_isDedicated && _showFinalMessage)
                Padding(
                  padding: const EdgeInsets.fromLTRB(24, 0, 24, 32),
                  child: SizedBox(
                    width: double.infinity,
                    child: ElevatedButton.icon(
                      onPressed: widget.onComplete,
                      icon: const Icon(Icons.arrow_forward_rounded, size: 16),
                      label: const Text('Enter Tribute',
                          style: TextStyle(fontWeight: FontWeight.w600, fontSize: 16)),
                      style: ElevatedButton.styleFrom(
                        backgroundColor: TributeColor.golden,
                        foregroundColor: TributeColor.charcoal,
                        padding: const EdgeInsets.symmetric(vertical: 16),
                        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(14)),
                      ),
                    ),
                  ),
                ),
            ]),
          ],
        );
      },
    );
  }

  Widget _preOfferingContent() {
    return Column(children: [
      AnimatedOpacity(
        opacity: _showVerse ? 1.0 : 0.0,
        duration: const Duration(milliseconds: 600),
        child: Column(children: [
          const Text('Your Tribute',
              style: TextStyle(fontSize: 22, fontWeight: FontWeight.w700, color: TributeColor.warmWhite)),
          const SizedBox(height: 8),
          Text('Everything you\u2019ve set \u2014 offered to God.',
              textAlign: TextAlign.center,
              style: TextStyle(fontSize: 15, color: Colors.white.withValues(alpha: 0.5))),
        ]),
      ),
      const SizedBox(height: 24),
      _habitTile(
        icon: Icons.volunteer_activism,
        name: 'Daily Gratitude',
        detail: HabitCategory.gratitude.defaultPurpose,
        accent: TributeColor.golden,
        isShowing: _showGratitudeTile,
        isGlowing: _tilesGlow,
      ),
      const SizedBox(height: 14),
      _habitTile(
        icon: _categoryIcon(),
        name: widget.habitName,
        detail: widget.purposeStatement.isNotEmpty
            ? widget.purposeStatement
            : widget.habitCategory.defaultPurpose,
        accent: _accentColor,
        isShowing: _showHabitTile,
        isGlowing: _tilesGlow,
      ),
      const SizedBox(height: 16),
      AnimatedOpacity(
        opacity: _showHabitTile ? 1.0 : 0.0,
        duration: const Duration(milliseconds: 500),
        child: _microMilestoneSection(),
      ),
      const SizedBox(height: 16),
      AnimatedOpacity(
        opacity: _showVerse ? 1.0 : 0.0,
        duration: const Duration(milliseconds: 600),
        child: Column(children: [
          Text(
            '\u201CTherefore, I urge you, brothers and sisters, in view of God\u2019s mercy, to offer your bodies as a living sacrifice, holy and pleasing to God \u2014 this is your true and proper worship.\u201D',
            textAlign: TextAlign.center,
            style: TextStyle(
              fontSize: 14, fontStyle: FontStyle.italic, height: 1.6,
              color: TributeColor.softGold.withValues(alpha: 0.6),
            ),
          ),
          const SizedBox(height: 6),
          Text('Romans 12:1',
              style: TextStyle(fontSize: 12, color: TributeColor.golden.withValues(alpha: 0.5))),
          if (DateTime.now().weekday != 7) ...[
            const SizedBox(height: 20),
            Container(
              padding: const EdgeInsets.all(12),
              decoration: BoxDecoration(
                color: TributeColor.golden.withValues(alpha: 0.06),
                borderRadius: BorderRadius.circular(12),
                border: Border.all(color: TributeColor.golden.withValues(alpha: 0.15), width: 0.5),
              ),
              child: Text(
                'Starting mid-week? No problem. Your first full Sunday cycle begins this coming Sunday.',
                textAlign: TextAlign.center,
                style: TextStyle(fontSize: 12, color: Colors.white.withValues(alpha: 0.6), height: 1.5),
              ),
            ),
          ],
        ]),
      ),
    ]);
  }

  Widget _postOfferingContent() {
    final b = _breatheController.value;
    return Column(children: [
      const SizedBox(height: 40),
      Stack(alignment: Alignment.center, children: [
        if (_showPulse) _pulseView(),
        AnimatedOpacity(
          opacity: _showFinalMessage ? 1.0 : 0.0,
          duration: const Duration(milliseconds: 600),
          child: AnimatedScale(
            scale: _showFinalMessage ? 1.0 : 0.92,
            duration: const Duration(milliseconds: 600),
            curve: Curves.easeOutBack,
            child: Column(children: [
              Container(
                width: 90, height: 90,
                decoration: BoxDecoration(
                  shape: BoxShape.circle,
                  gradient: RadialGradient(colors: [
                    TributeColor.golden.withValues(alpha: 0.4),
                    TributeColor.golden.withValues(alpha: 0.1),
                  ]),
                ),
                child: Transform.scale(
                  scale: 1.0 + b * 0.08,
                  child: const Icon(Icons.volunteer_activism, size: 38, color: TributeColor.golden),
                ),
              ),
              const SizedBox(height: 20),
              const Text('Your tribute is set.',
                  style: TextStyle(fontSize: 18, fontWeight: FontWeight.w700, color: TributeColor.warmWhite)),
              const SizedBox(height: 8),
              Text(
                  widget.givenName != null
                      ? 'Remain steadfast in all you do, ${widget.givenName}.'
                      : 'Remain steadfast in all you do.',
                  style: const TextStyle(fontSize: 17, fontWeight: FontWeight.w600, color: TributeColor.softGold)),
            ]),
          ),
        ),
      ]),
      const SizedBox(height: 32),
      AnimatedOpacity(
        opacity: _showFinalMessage ? 1.0 : 0.0,
        duration: const Duration(milliseconds: 600),
        child: Column(children: [
          Text(
            '\u201CThe steadfast love of the Lord never ceases; his mercies never come to an end; they are new every morning.\u201D',
            textAlign: TextAlign.center,
            style: TextStyle(
              fontSize: 14, fontStyle: FontStyle.italic, height: 1.6,
              color: TributeColor.softGold.withValues(alpha: 0.5),
            ),
          ),
          const SizedBox(height: 6),
          Text('Lamentations 3:22\u201323',
              style: TextStyle(fontSize: 12, color: TributeColor.golden.withValues(alpha: 0.4))),
          const SizedBox(height: 24),
          Container(
            padding: const EdgeInsets.all(16),
            decoration: BoxDecoration(
              color: TributeColor.cardBackground,
              borderRadius: BorderRadius.circular(16),
              border: Border.all(color: TributeColor.cardBorder, width: 0.5),
            ),
            child: Column(children: [
              const Icon(Icons.groups_rounded, size: 28, color: TributeColor.golden),
              const SizedBox(height: 10),
              const Text('Want to invite a few people to walk with you?',
                  textAlign: TextAlign.center,
                  style: TextStyle(fontSize: 14, fontWeight: FontWeight.w600, color: TributeColor.warmWhite)),
              const SizedBox(height: 4),
              Text('In Tribute you can start a Prayer Circle and do this together.',
                  textAlign: TextAlign.center,
                  style: TextStyle(fontSize: 12, color: Colors.white.withValues(alpha: 0.5))),
            ]),
          ),
        ]),
      ),
    ]);
  }

  Widget _microMilestoneSection() {
    final weekday = DateTime.now().weekday;
    final daysLeft = weekday == 7 ? 6 : (6 - weekday + 1);
    if (daysLeft <= 0) return const SizedBox.shrink();

    final gratitudeLine = '$daysLeft day${daysLeft == 1 ? '' : 's'} of gratitude';

    final String customLine;
    switch (widget.trackingType) {
      case HabitTrackingType.timed:
        final mins = (widget.dailyTarget * daysLeft).toInt();
        final unit = widget.targetUnit.isNotEmpty ? widget.targetUnit : 'minutes';
        customLine = '$mins $unit of ${widget.habitName.toLowerCase()}';
      case HabitTrackingType.count:
        final count = (widget.dailyTarget * daysLeft).toInt();
        final unit = widget.targetUnit.isNotEmpty ? widget.targetUnit : 'completed';
        customLine = '$count $unit of ${widget.habitName.toLowerCase()}';
      case HabitTrackingType.checkIn:
        customLine = '$daysLeft day${daysLeft == 1 ? '' : 's'} of ${widget.habitName.toLowerCase()}';
      case HabitTrackingType.abstain:
        customLine = '$daysLeft clean day${daysLeft == 1 ? '' : 's'}';
    }

    return Container(
      padding: const EdgeInsets.all(14),
      decoration: BoxDecoration(
        color: TributeColor.golden.withValues(alpha: 0.04),
        borderRadius: BorderRadius.circular(12),
        border: Border.all(color: TributeColor.golden.withValues(alpha: 0.12), width: 0.5),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          const Text('If you hit your targets this week:',
              style: TextStyle(fontSize: 13, fontWeight: FontWeight.w600, color: TributeColor.softGold)),
          const SizedBox(height: 10),
          _milestoneLine(gratitudeLine),
          if (widget.habitName.isNotEmpty) _milestoneLine(customLine),
        ],
      ),
    );
  }

  Widget _milestoneLine(String text) {
    return Padding(
      padding: const EdgeInsets.only(bottom: 6),
      child: Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
        const Icon(Icons.auto_awesome, size: 11, color: TributeColor.golden),
        const SizedBox(width: 8),
        Expanded(
          child: Text(text,
              style: TextStyle(fontSize: 13, color: Colors.white.withValues(alpha: 0.6), height: 1.4)),
        ),
      ]),
    );
  }

  Widget _habitTile({
    required IconData icon,
    required String name,
    required String detail,
    required Color accent,
    required bool isShowing,
    required bool isGlowing,
  }) {
    return AnimatedOpacity(
      opacity: isShowing ? 1.0 : 0.0,
      duration: const Duration(milliseconds: 400),
      child: AnimatedSlide(
        offset: isShowing ? Offset.zero : const Offset(0, 0.2),
        duration: const Duration(milliseconds: 400),
        child: AnimatedContainer(
          duration: const Duration(milliseconds: 400),
          padding: const EdgeInsets.all(16),
          decoration: BoxDecoration(
            color: isGlowing ? accent.withValues(alpha: 0.08) : TributeColor.cardBackground,
            borderRadius: BorderRadius.circular(16),
            border: Border.all(
              color: isGlowing ? accent.withValues(alpha: 0.3) : TributeColor.cardBorder,
              width: 0.5,
            ),
          ),
          child: Row(children: [
            Container(
              width: 48, height: 48,
              decoration: BoxDecoration(
                shape: BoxShape.circle,
                gradient: RadialGradient(colors: [
                  accent.withValues(alpha: isGlowing ? 0.5 : 0.2),
                  accent.withValues(alpha: isGlowing ? 0.2 : 0.06),
                ]),
              ),
              child: Icon(icon, size: 22, color: accent),
            ),
            const SizedBox(width: 14),
            Expanded(
              child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
                Text(name,
                    style: const TextStyle(fontSize: 17, fontWeight: FontWeight.w600, color: TributeColor.warmWhite)),
                const SizedBox(height: 4),
                Text(detail,
                    style: TextStyle(fontSize: 12, color: Colors.white.withValues(alpha: 0.5))),
              ]),
            ),
            Icon(
              isGlowing ? Icons.check_circle_rounded : Icons.circle_outlined,
              size: 22,
              color: isGlowing ? TributeColor.golden : Colors.white.withValues(alpha: 0.15),
            ),
          ]),
        ),
      ),
    );
  }

  Widget _pulseView() {
    return SizedBox(
      width: 200, height: 200,
      child: Stack(
        alignment: Alignment.center,
        children: List.generate(4, (i) => _PulseRing(delay: i * 200)),
      ),
    );
  }
}

class _PulseRing extends StatefulWidget {
  final int delay;
  const _PulseRing({required this.delay});

  @override
  State<_PulseRing> createState() => _PulseRingState();
}

class _PulseRingState extends State<_PulseRing> with SingleTickerProviderStateMixin {
  late final AnimationController _ctrl;
  late final Animation<double> _scale;
  late final Animation<double> _opacity;

  @override
  void initState() {
    super.initState();
    _ctrl = AnimationController(vsync: this, duration: const Duration(milliseconds: 1500));
    _scale = Tween<double>(begin: 0.2, end: 3.0).animate(CurvedAnimation(parent: _ctrl, curve: Curves.easeOut));
    _opacity = Tween<double>(begin: 0.6, end: 0.0).animate(CurvedAnimation(parent: _ctrl, curve: Curves.easeOut));
    Future.delayed(Duration(milliseconds: widget.delay), () {
      if (mounted) _ctrl.repeat();
    });
  }

  @override
  void dispose() {
    _ctrl.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return AnimatedBuilder(
      animation: _ctrl,
      builder: (context, _) => Transform.scale(
        scale: _scale.value,
        child: Opacity(
          opacity: _opacity.value,
          child: Container(
            width: 60, height: 60,
            decoration: BoxDecoration(
              shape: BoxShape.circle,
              gradient: RadialGradient(colors: [
                TributeColor.golden.withValues(alpha: 0.4),
                TributeColor.softGold.withValues(alpha: 0.15),
                Colors.transparent,
              ]),
            ),
          ),
        ),
      ),
    );
  }
}
